### PR TITLE
Update ReflectionClassDeclaration.java

### PR DESCRIPTION
**Pull request by stefanleh**
Add a small optimization when there is no method to to check for being applicable.

### CODE REVIEW

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionClassDeclaration.java [hunk 1]
```diff
@@ -154,6 +154,10 @@ public String getQualifiedName() {
                 methods.add(ref.getCorrespondingDeclaration());
             }
         }
+        // when empty there is no sense in trying to find the most applicable
```
Comment (ftomassetti) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionClassDeclaration.java:157]: Yes, that is correct but I wonder if this optimization is useful

Comment (stefanleh) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionClassDeclaration.java:157]: I was debugging my other resultion issue and came across that part which got entered a lot even if there was an empty list. Maybe its not a big performance gain but it help me with debugging ;)

Comment (ftomassetti) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionClassDeclaration.java:157]: Ok then :) Merging as tests complete

**Comment by ftomassetti** [comment]
+1

**Comment by ftomassetti** [comment]
Superseded by #1788 therefore I am closing this one

**Pull request closed without merging**